### PR TITLE
2327 by Inlead: Various permission improvements.

### PR DESCRIPTION
**Pull request by inlead**

### CODE REVIEW

File: modules/ding_permissions/ding_permissions.module [hunk 1]
```diff
@@ -1011,3 +1013,14 @@ function ding_permissions_secure_permissions($role) {
     return $permissions[$role];
   }
 }
+
+/**
+ * Implements hook_page_alter().
```
Comment (cableman) [inline modules/ding_permissions/ding_permissions.module]: A comment about why this change in the page is need would be nice.

Comment (bozeak) [inline modules/ding_permissions/ding_permissions.module]: There was a problem with toolbars (core toolbar and admin_menu toolbar) when the user with role "administrator" and uid=1 was logged in, both toolbars was displayed. Through standard permissions system it was not possible to disable core toolbar for administrator role, we were forced to unset this on page build.

Comment (cableman) [inline modules/ding_permissions/ding_permissions.module]: Would it be not be possible to just use one of the toolbars? Simply disabling the one you don't want to use?

Comment (guddo) [inline modules/ding_permissions/ding_permissions.module]: @cableman That's a decision by the business - you can [read more about it in the ticket tracker](http://platform.dandigbib.org/issues/2206).

Comment (cableman) [inline modules/ding_permissions/ding_permissions.module]: Still would like a comment in the code about this, so we also knowns why it's here in the future.

Comment (holt83) [inline modules/ding_permissions/ding_permissions.module]: I don't see it as a big problem that uid=1 has two toolbars. As administrator you can easily just disable one of the modules.

We removed hook_page_alter() in http://platform.dandigbib.org/issues/2184 in favor of handling it in hook_secure_permission. I see no reason to reintroduce this hook just for uid=1. If permissions is set correctly up for all the roles, there shouldn't really be any problem.

**Comment by cableman** [comment]
Reviewed and awaiting release.
